Onwubiko-Chibuike/main.js

```diff
@@ -0,0 +1,367 @@
+    if(output.totalResults > 1) {
+        wordings = " Search results for ";
+    } else if(output.totalResults == 1 ) {
+        wordings = " Search result for ";
+    } else {
+        wordings = "0 Search result for";
+    }
+    searchHeader.textContent = wordings + search.value;
+    main.appendChild(searchHeader);
+    for (let i = 0; i < output.articles.length; i++) {
+        let searchResult = document.createElement('div');
+        let a = document.createElement('a');
+        let p = document.createElement('p');
+        let searchResultLink = searchResult.appendChild(a);
+        let searchResultContent = searchResult.appendChild(p);
+        a.setAttribute('target', 'blank');
+        searchResult.setAttribute('class', 'search-results');
+        searchResultLink.textContent = output.articles[i].title;
+        searchResultLink.href = output.articles[i].url;
+        searchResultContent.textContent = output.articles[i].description;
+        main.appendChild(searchResult);
+    }
+    return;
+}
+
+function catchError(error) {
+    console.log('looks like there was a problem: \n' + error );
+    let errorParagraph = document.createElement('p');
+    errorParagraph.textContent = "Kindly Check your internet connection";
+    //main.insertBefore(errorParagraph, newsContainer);
+    //errorParagraph.setAttribute('class', 'all__font-size');
+}
+
+function makeRequest(req,displayOutput) {
+    fetch(req)
+    .then(validateAndReturnResponse)
+    .then(displayOutput)
+    .catch(catchError)
+}
+
+makeRequest(req, displayOutput);
+makeRequest(sportsRequest, displaySportsOutput);
+makeRequest(technologyRequest, displayTechnologyOutput);
+
+
+window.addEventListener('load', getFromLocalStorage('topNews',displayOutput));
+window.addEventListener('load', getFromLocalStorage('sportsNews', displaySportsOutput));
+window.addEventListener('load', getFromLocalStorage('techNews', displayTechnologyOutput));
+
+//Making a search request for news
+window.addEventListener('DOMContentLoaded', getSearch);
+function getSearch() {
+    let getSearchResult = document.getElementById('form');
+    getSearchResult.addEventListener('submit', makeSearchRequest, false);
+    return;
+}
+function removePreviousSearchResult() {
+    let query = document.querySelectorAll('.search-results');
+    query.forEach( (q) => {q.remove();});
+    searchHeader.remove();
+    return;
+}
+
+//Searching for news 
+function makeSearchRequest(e) {
+    e.preventDefault();
+
+    removePreviousSearchResult();
+    
+    timing();
+
+    newsContainer.style.display = 'none';
+    displayTech.style.display = 'none';
+    displaySport.style.display = 'none';
+
+    let formData = new FormData(document.getElementById('form')); 
+
+    var data = new URLSearchParams();
+    for (let pair of formData) {
+        data.append(pair[0],pair[1]);
+    }
+
+    data.append('apikey', apikeys);
+    data.toString();
+
+    let searchUrl = 'https://newsapi.org/v2/top-headlines?' + data;
```

Comment:
```suggestion
    let searchUrl = 'https://newsapi.org/v2/everything?' + data;
```

Reply:
changed it and i am getting an error 405 anytime i try searching online

Reply:
Solved sir.

Reply:
done

Reply:
Done